### PR TITLE
GH-284: Update Apache Arrow website references to Jira

### LINKED ISSUE #284: Update Apache Arrow website references to Jira
### Describe the enhancement requested

The Apache Arrow website has references to Jira which should be updated to point to GitHub.

### Component(s)

Website

### PR DESCRIPTION
**Pull request by AlenkaF**
This PR is part of the work to update the documentation as we are moving from Jira issue tracker to GitHub issues:
https://github.com/apache/arrow/issues/14822

closes #284

### CODE REVIEW

File: community.md [hunk 1]
```diff
@@ -75,18 +74,14 @@ their own tags (for example,
...
-
-If you know your question is actually a bug report or feature request, we
-encourage you to go straight to JIRA. If you're not sure, feel free to make a
-GitHub issue to ask your question and we can triage/redirect your question
-there.
+We use GitHub issues as a way to ask questions and engage with the Arrow developer
+community but also for maintaining a queue of development work and as the public
+record for work on the project. We use the mailing lists for development discussions,
+so to keep things in a single place, we prefer not to have lengthy discussions on
+GitHub issues.
```
Comment (rok) [inline community.md:81]:
```suggestion
We use GitHub issues as a way to ask questions and engage with the Arrow developer
community and for maintaining a queue of development work and as the public
record of work on the project. We use the mailing lists for development discussions,
where a lengthy discussions is required.
```

**Review by raulcd — APPROVED** [review]
LGTM, there are other PRs on the arrow repo to also update the new contributors guide and the developers section but I am happy to merge this

**Comment by rok** [comment]
@raulcd sorry, I just updated the header link and some metadata as well. If you're ok with that I'll merge.

**Comment by raulcd** [comment]
Thanks @rok looks good to me but as this is a GitHub issue instead of a JIRA issue we might have to fix this https://github.com/apache/arrow-site/issues/285 first? I forgot that we still had to update the merge script on this repo

**Comment by raulcd** [comment]
It seems sometime we merge manually on this repo, we can probably merge manually and close the issue manually and update the merge_script in the future

**Pull request merged**